### ISSUE TITLE
Update wiki or GUI with legend to explain/show sample of rename options

### ISSUE THREAD

**Issue by CollinChaffin** [Sonarr/Sonarr#2426]
Would it be possible to add a simple legend to actually show/demonstrate what each of these will generate without having to parse the source code?

Some of these options even when pasted into the "live preview" which appears to be the only current way to even know what they generate, do not even display what the end result would be (ie. Quality Real, etc.).  TIA!

# Rename Options

{MediaInfo Audio}
{MediaInfo AudioChannels}
{MediaInfo AudioCodec}
{MediaInfo Full}
{MediaInfo Simple}
{MediaInfo Video}
{MediaInfo VideoBitDepth}
{MediaInfo VideoCodec}
{Series CleanTitle}
{Series Title}
{Series TitleThe}
{Original Filename}
{Original Title}
{Quality Full}
{Quality Proper}
{Quality Real}
{Quality Title}
{Release Group}

**Comment by markus101** [comment]
This is something that has been done in the v3 work.

**Comment by CollinChaffin** [comment]
Could you possibly link a pointer to it here?

**Comment by markus101** [comment]
![image](https://user-images.githubusercontent.com/413544/36622770-2fd4b06c-18b4-11e8-8e7a-dcf4165975b3.png)
![image](https://user-images.githubusercontent.com/413544/36622784-4b16c554-18b4-11e8-85a3-04ae5152bd85.png)

Looks like it's missing `quality proper` and `quality real` which would either be nothing if it's not a proper/real or `proper`/`real` and overall could use some de-duping and moving the separator to a select.

**Comment by CollinChaffin** [comment]
Excellent, thanks Mark!  I'm brand new to Sonarr and always written my own sorters but just don't have the time lately to keep up with changes needed for all the other APIs, etc. so am excited to give Sonarr a try!  So let me ask you since the standard naming convention in all the scene releases is this:

````
The.Show.s01e01.Awesome.Happenings.PROPER.720p.HDTV.x264.EVOLVE <----CORRECT/STANDARD
````

Yet from what I can tell the default it seems in Sonarr is this:
````
The.Show.s01e01.Awesome.Happenings.PROPER.HDTV.720p.x264.EVOLVE <----INCORRECT/NON-STANDARD
````

So my question is, how do I place the DETECTED RESOLUTION before the TYPE TAG (WebDL/BluRay/HDTV)?  I'm sure this is possible and it's probably right in front of me and I'm just having a brain-fart in my old age!  Thanks again for taking the time to help!

**Comment by markus101** [comment]
You can change the format Sonarr uses for quality on the quality settings page.

**Comment by CollinChaffin** [comment]
Hey Mark I know this is closed but forgot to ask one last related question regarding restriction tags.  This is super-common so hopefully there is a way I'm just not seeing but given the above pic you posted, is there a way to always include the tag (that forced the match, if one exists)?  In other words, the very common AMZN tag....is there any way (or is there a pending feature request) to add `{FIRSTMATCHEDTAG}` into the renaming to allow for instance, anyone's `AMZN` or `NF.WEB` to be added (and accepted) as part of the filename?

Also, as it relates to this.....if not, I'm worried that because my library DOES comply with all current Plex naming standards (ie. `Lost.s01e01.Cool.Ep.Name.1080p.WebDL.AMZN.x264`).  But, now I fear when I point Sonarr at a huge library with this current (working) naming set up for Plex, that Sonarr will use it's current rename template and destroy my library and remove all my Amazon release tags, etc.  So the 2nd part of this question is, outside of adding tags, is there a way to safely import into Sonarr what I know it will READ successfully, but WITHOUT it then, after adding, changing any filenames?

Thanks again for all your help!

**Comment by markus101** [comment]
There isn't a way to add additional keywords to file names, the only way would be to use the original title/filename, but that comes with it's own issues. At this time pulling out keywords out of the original title/filename isn't something we have planned, though that may be advantageous for https://github.com/Sonarr/Sonarr/issues/385 depending if upgrading to a more preferred release is included.

Sonarr won't automatically rename any existing files, it'd only do that during renaming. If Sonarr was told to organize and rename the files in one or more series (from Series Editor) or to a single series or a season of that series (from Series Details) then it'd rename the files.

**Comment by CollinChaffin** [comment]
What about upon initial setup and importing from disk?  Can I in effect toggle OFF all file renaming, add all my library, then toggle file renaming on and only have it apply to brand new episodes?  Or, after the toggle on and everything added, will it then at any time (under automatic) periodicatlly scan and then still want to forcibly rename my files, that (other than the AMZN, NF, etc. denotation) are already in perfect and consistent format?

In other words, my library is perfect other than these Amazon and Netflix tags that like many I have seen post, do NOT want to lose......but....and it's a big but.....I may be able to live with NEW episodes for the time being missing those tags (or if I have to add them all manually) since Sonarr it looks like does literally everything else I need it to.

As usual, thanks again for taking the time!
-C

**Comment by markus101** [comment]
If they're already within the series folder then they are not renamed at import time, only files imported from outside the series folder are renamed when importing (if enabled). Your concern is not an issue.

**Comment by CollinChaffin** [comment]
EDIT: Note what I was asking about utilizing TAG from the user-set Restrictions tags.....so I have a tag with "AMZN" as the data field, and if matched what I was asking is since so many utilize that (it seems it is THE way to currently denote downloading AMZN or NF WebDL's), it would be super-cool to then give us all a way to use the fact that the match occurred and simply use a new variable called {FIRSTMATCHEDTAG} or something and just pass it directly into the filename - which in this case would then name all my AMZN or NF releases accordingly!

**Comment by CollinChaffin** [comment]
Ok awesome so #2 is good - my existing library AMZN tags will then not be changed since they are all existing episodes but please put some thought into the other request/idea with all the posts I've seen on AMZN and similar quality/source tagged releases!  TIA!
